### PR TITLE
CI: fetch upgrade logs to the archive

### PR DESCRIPTION
**Pull request by Jaakko-Os**

### CODE REVIEW

**Review by fmuyassarov — COMMENTED** [review]
You could try to fetch the logs only for the upgrade CI. It just needs a conditional statement. 
```
if [[ "${TESTS_FOR}" == "feature_tests_upgrade* ]]
then
   ...fetch the logs...
fi
```

**Comment by Jaakko-Os** [comment]
> You could try to fetch the logs only for the upgrade CI. It just needs a conditional statement.
> 
> ```
> if [[ "${TESTS_FOR}" == "feature_tests_upgrade* ]]
> then
>    ...fetch the logs...
> fi
> ```

Good comment! Done

**Comment by Jaakko-Os** [comment]
> > You could try to fetch the logs only for the upgrade CI. It just needs a conditional statement.
> > ```
> > if [[ "${TESTS_FOR}" == "feature_tests_upgrade* ]]
> > then
> >    ...fetch the logs...
> > fi
> > ```
> 
> Good comment! Done

/tmp/run_fetch_logs.sh: line 57: TESTS_FOR: unbound variable

going back for the non env var dependent implementation

**Comment by fmuyassarov** [comment]
> > > You could try to fetch the logs only for the upgrade CI. It just needs a conditional statement.
> > > ```
> > > if [[ "${TESTS_FOR}" == "feature_tests_upgrade* ]]
> > > then
> > >    ...fetch the logs...
> > > fi
> > > ```
> > 
> > 
> > Good comment! Done
> 
> /tmp/run_fetch_logs.sh: line 57: TESTS_FOR: unbound variable
> 
> going back for the non env var dependent implementation

You need to make sure that you fetch the value from the Jenkins pipeline. The variable is available in Jenkins pipeline https://github.com/metal3-io/project-infra/blob/0a813784186d1ea1812c2e83175da34c3e88e97f/jenkins/jobs/integration_tests.pipeline#L58 
You can see an [example](https://github.com/metal3-io/project-infra/blob/7f4d6420f629d7cda27e1b13edc32b770b6ef7fb/jenkins/scripts/files/run_integration_tests.sh#L124) that is using TESTS_FOR variable.

File: jenkins/scripts/fetch_logs.sh [hunk 1]
```diff
@@ -13,7 +13,7 @@ set -eu
 CI_DIR="$(dirname "$(readlink -f "${0}")")"
 DISTRIBUTION="${DISTRIBUTION:-ubuntu}"
 BUILD_TAG="${BUILD_TAG:-logs_integration_tests}"
-
+TESTS_FOR="${TESTS_FOR}"
```
Comment (maelk) [inline jenkins/scripts/fetch_logs.sh:16]: that is not doing anything

File: jenkins/scripts/files/run_fetch_logs.sh [hunk 2]
```diff
@@ -6,6 +6,8 @@ set -u
 LOGS_TARBALL=${1:-container_logs.tgz}
 LOGS_DIR="${2:-logs}"
 DISTRIBUTION="${3:-ubuntu}"
+TESTS_FOR="${4:-noupgrade}"
```
Comment (maelk) [inline jenkins/scripts/files/run_fetch_logs.sh:9]: you could also default to an empty string. "noupgrade" does not appear anywhere, so could raise questions.

**Review by maelk — APPROVED** [review]

**Comment by maelk** [comment]
some nits that could be fixed before merging

File: jenkins/scripts/fetch_logs.sh (hunk 1 above)
Comment (Jaakko-Os) [reply jenkins/scripts/fetch_logs.sh]: removed

File: jenkins/scripts/files/run_fetch_logs.sh (hunk 2 above)
Comment (Jaakko-Os) [reply jenkins/scripts/files/run_fetch_logs.sh]: done

**Pull request merged**